src/psyclone/tests/psyir/backend/psyir_openmp_test.py

```diff
@@ -0,0 +1,239 @@
+    is created correctly.
+    '''
+
+    from psyclone.transformations import OMPParallelTrans
+    from psyclone.tests.psyclone_test_utils import get_invoke
+
+    _, invoke = get_invoke("single_invoke.f90", "gocean1.0",
+                           idx=0)
+
+    omp = OMPParallelTrans()
+    omp_sched, _ = omp.apply(invoke.schedule[0])
+
+    # Now remove the GOKern (since it's not yet supported in the
+    # visitor pattern) and replace it with a simple assignment
+    replace_child_with_assignment(omp_sched[0])
+
+    # omp_sched is a GOInvokeSchedule, which is not yet supported.
+    # So only convert starting from the OMPParallelDirective
+    fvisitor = FortranWriter()
+    result = fvisitor(omp_sched[0])
+    correct = '''!$omp parallel
+  a=b
+!$omp end parallel'''
+    assert correct in result
+
+    cvisitor = CWriter()
+    # Remove newlines for easier RE matching
+    result = cvisitor(omp_sched[0])
+    correct = '''#pragma omp parallel
+{
+  a = b;
+}'''
+    result = cvisitor(omp_sched[0])
+    assert correct in result
+
+
+# ----------------------------------------------------------------------------
+def test_nemo_omp_do():
+    '''Tests if an OpenMP do directive in NEMO is handled correctly.
+    '''
+    # Generate fparser2 parse tree from Fortran code.
+    code = '''
+        module test
+        contains
+        subroutine tmp()
+          integer :: i, a
+          integer, dimension(:) :: b
+          do i = 1, 20, 2
+            a = 2 * i
+            b(i) = b(i) + a
+          enddo
+        end subroutine tmp
+        end module test'''
+    schedule = create_schedule(code)
+    from psyclone.transformations import OMPLoopTrans
+
+    # Now apply a parallel transform
+    omp_loop = OMPLoopTrans()
+    omp_loop.apply(schedule[0])
+
+    fvisitor = FortranWriter()
+    result = fvisitor(schedule)
+    correct = '''!$omp do schedule(static)
+      do i = 1, 20, 2
+        a=2 * i
+        b(i)=b(i) + a
+      enddo
+!$omp end do'''
+    assert correct in result
+
+    cvisitor = CWriter()
+    result = cvisitor(schedule[0])
+    correct = '''#pragma omp do schedule(static)
+{
+  for(i=1-1; i<20; i+=2)
+  {
+    a = (2 * i);
+    b[i] = (b[i] + a);
+  }
+}'''
+    assert correct in result
+
+
+# ----------------------------------------------------------------------------
+def test_gocean_omp_do():
+    '''Test that an OMP PARALLEL directive in a 'classical' API (gocean here)
+    is created correctly.
+    '''
+
+    from psyclone.transformations import OMPLoopTrans
+    from psyclone.tests.psyclone_test_utils import get_invoke
+
+    _, invoke = get_invoke("single_invoke.f90", "gocean1.0",
+                           idx=0)
+    omp = OMPLoopTrans()
+    omp_sched, _ = omp.apply(invoke.schedule[0])
+
+    # Now remove the GOKern (since it's not yet supported in the
+    # visitor pattern) and replace it with a simple assignment.
+    # While this is invalid usage of OMP (omp must have a loop,
```

Comment:
Unfinished comment?

Reply:
Done ("# visitor" was just a left-over when I reformatted the line breaks).